x-pack/plugins/maps/public/ems_autosuggest/ems_autosuggest.ts

```diff
@@ -0,0 +1,204 @@
+
+  if (sampleValuesConfig.sampleValuesColumnName) {
+    matches.push(...suggestByName(sampleValuesConfig.sampleValuesColumnName));
+  }
+
+  if (sampleValuesConfig.sampleValues && sampleValuesConfig.sampleValues.length) {
+    if (sampleValuesConfig.emsLayerIds && sampleValuesConfig.emsLayerIds.length) {
+      matches.push(
+        ...(await suggestByEMSLayerIds(
+          sampleValuesConfig.emsLayerIds,
+          sampleValuesConfig.sampleValues
+        ))
+      );
+    } else {
+      matches.push(...suggestByValues(sampleValuesConfig.sampleValues));
+    }
+  }
+
+  const uniqMatches: UniqueMatch[] = matches.reduce((accum: UniqueMatch[], match) => {
+    const found = accum.find((m) => {
+      return m.config.layerId === match.layerId && m.config.field === match.layerId;
+    });
+
+    if (found) {
+      found.count += 1;
+    } else {
+      accum.push({
+        config: match,
+        count: 1,
+      });
+    }
+
+    return accum;
+  }, []);
+
+  uniqMatches.sort((a, b) => {
+    return b.count - a.count;
+  });
+
+  return uniqMatches.length ? uniqMatches[0].config : null;
+}
+
+function suggestByName(columnName: string): IEMSTermJoinConfig[] {
+  const matches = wellKnownColumnNames.filter((wellknown) => {
+    return columnName.match(wellknown.regex);
+  });
+
+  return matches.map((m) => {
+    return m.emsConfig;
+  });
+}
+
+function suggestByValues(values: Array<string | number>): IEMSTermJoinConfig[] {
+  const matches = wellKnownColumnFormats.filter((wellknown) => {
+    for (let i = 0; i < values.length; i++) {
+      const value = values[i].toString();
+      if (!value.match(wellknown.regex)) {
+        return false;
+      }
+    }
+    return true;
+  });
+
+  return matches.map((m) => {
+    return m.emsConfig;
+  });
+}
+
+function existsInEMS(emsJson: any, emsFieldId: string, sampleValue: string): boolean {
+  for (let i = 0; i < emsJson.features.length; i++) {
+    const emsFieldValue = emsJson.features[i].properties[emsFieldId].toString();
+    if (emsFieldValue.toString() === sampleValue) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function matchesEmsField(emsJson: any, emsFieldId: string, sampleValues: Array<string | number>) {
+  for (let j = 0; j < sampleValues.length; j++) {
+    const sampleValue = sampleValues[j].toString();
+    if (!existsInEMS(emsJson, emsFieldId, sampleValue)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+async function getMatchesForEMSLayer(
+  emsLayerId: string,
+  sampleValues: Array<string | number>
+): Promise<IEMSTermJoinConfig[]> {
+  const fileLayers: FileLayer[] = await getEmsFileLayers();
+  const emsFileLayer: FileLayer | undefined = fileLayers.find((fl: FileLayer) =>
+    fl.hasId(emsLayerId)
+  );
+
+  if (!emsFileLayer) {
+    return [];
+  }
+
+  const emsFields = emsFileLayer.getFields();
+  const url = emsFileLayer.getDefaultFormatUrl();
+
+  try {
+    const emsJson = await fetchGeoJson(
```

Comment:
Should we put these geojson files into some kind of cache so they are not re-fetched everything time the method is run?

Reply:
discussed offline. Will push cashing logic in the ems-client. This will also simplify the geojson-fetching in general, pushing format-conversion (if any) into the ems-client as well. cf. https://github.com/elastic/ems-client/issues/60